src/csharp/GeneratorParams.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI
+{
+    public class GeneratorParams : IDisposable
+    {
+        private IntPtr _generatorParamsHandle;
+        private bool _disposed = false;
+
+        public GeneratorParams(Model model)
+        {
+            Result.VerifySuccess(NativeMethods.OgaCreateGeneratorParams(model.Handle, out _generatorParamsHandle));
+        }
+
+        internal IntPtr Handle { get { return _generatorParamsHandle; } }
+
+        public void SetMaxLength(ulong maxLength)
+        {
+            Result.VerifySuccess(NativeMethods.OgaGeneratorParamsSetMaxLength(_generatorParamsHandle, (UIntPtr)maxLength));
+        }
+
+        public void SetInputIDs(IReadOnlyCollection<int> inputIDs, ulong sequenceLength, ulong batchSize)
+        {
+            int[] inputIDsArray = inputIDs.ToArray();
```

Comment:
This produces a copy + garbage collected array on every call.

Here is my suggestion, the inputIDs must be in a contagious memory for this API call. Therefore, let's change the input parameter to ReadOnlySpan<int>.

Then, you can obtain a pointer to its buffer, pin it and feed directly to the native call.

here is an example:
[https://github.com/microsoft/onnxruntime/blob/main/csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs#L783](https://github.com/microsoft/onnxruntime/blob/main/csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs#L783)